clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/ShareConsumerTest.java

```diff
@@ -2000,6 +1998,29 @@ public void testShareConsumerAfterCoordinatorMovement() throws Exception {
         verifyShareGroupStateTopicRecordsProduced();
     }
 
+    @ClusterTest
+    public void testDeliveryCountNotIncreaseAfterSessionClose() {
```

Comment:
Please can we have another test for explicit acknowledgement. In this case, for closing a share session normally, there is a difference between records which have been delivery and acknowledged by the application, and those which are released automatically as a result of the share session close. So, the test case could:
* Create a new share consumer setting explicit acknowledgement mode
* Fetch 2 records
* Acknowledge the first record with AcknowledgeType.RELEASE
* Commit the acknowledgement
* Close the consumer
* Create a new consumer and fetch the 2 records again
The delivery count for the first record should be 2, while the delivery count for the second record should be 1. This is because we know the first record was delivered, but the second we do not.